Sources/SwiftCrossUI/Views/Menu.swift

```diff
@@ -71,27 +74,45 @@ public struct Menu: TypeSafeView {
         size.x = buttonWidth ?? size.x
 
         let content = resolve().content
-        backend.updateButton(
-            widget,
-            label: label,
-            action: {
-                let menu = backend.createPopoverMenu()
+        switch backend.menuImplementationStyle {
+            case .dynamicPopover:
+                backend.updateButton(
+                    widget,
+                    label: label,
+                    action: {
+                        let menu = backend.createPopoverMenu()
+                        children.menu = menu
+                        backend.updatePopoverMenu(
+                            menu,
+                            content: content,
+                            environment: environment
+                        )
+                        backend.showPopoverMenu(menu, at: SIMD2(0, size.y + 2), relativeTo: widget)
+                        {
+                            children.menu = nil
+                        }
+                    },
+                    environment: environment
+                )
+
+                if !dryRun {
+                    backend.setSize(of: widget, to: size)
+                    children.updateMenuIfShown(
+                        content: content, environment: environment, backend: backend)
+                }
+            case .menuButton:
+                let menu = children.menu as? Backend.Menu ?? backend.createPopoverMenu()
                 children.menu = menu
```

Comment:
I believe this is redundant?

Reply:
This is a pattern I've used elsewhere before. The cast obscures it a little in this case, but the basic pattern is
```swift
let x = y.x ?? createX()
y.x = x
```
This ensures `x` exists and is attached to `y`. In this particular case it reduces the number of `createPopoverMenu` calls by caching the Menu object.